# Assignment Kennis en Data

## Notebook made by


|Naam|Studentnummer|
|-|-|
|Una|secret|
|Zubin|secret|
|Tijn|secret|
|Soufiane|secret|

## Peerreview

* Vul onderstaande dict/tabel in en run de cel. Klopt het? Zijn jullie het er allemaal mee eens?
    * Verander `1,2,3,4` door jullie studentnummers
    * `1: {1:.2,2:.3,3:.3,4:.2}` betekent dat student 1 zijn punt zo over de 4 studenten verdeelt
    * Let op, per student moeten de punten optellen tot 1!
    * De gegeven punten staan in de tabel op de rijen, de ontvangen punten in de kolommen.
    * met `cijfer_per_student` kan je bepalen welk cijfer ieder krijgt bij een gegeven groepscijfer.

## Toelichting

* De meeste opgaven worden automatisch nagekeken. Bij vrijwel alle opdrachten staan er een paar tests onder de opdracht, dit is voornamelijk om te zorgen dat je de juiste type output geeft. Dit zijn dus *NIET* alle tests, die komen er bij het graden nog bij.
* Elke vraag is 1 punt waard, tenzij anders aangegeven. Soms is die punt onderverdeeld in deelpunten, maar niet altijd. 

## Voor het inleveren!

* Pas niet de cellen aan, vooral niet die je niet kunt editen. Dit levert problemen op bij nakijken. Twijfel je of je per ongeluk iets hebt gewijzigd, kopieer dan bij inleveren je antwoorden naar een nieuw bestand, zodat het niet fout kan gaan.

* Zorg dat de code goed runt van boven naar beneden, verifieer dat door boven in Kernel -> Restart & Run All uit te voeren

## Na het inleveren!

* Het gebeurt erg vaak dat mensen een "leeg bestand" inleveren. Vaak een andere versie van de opgave die nog ergens op je computer rondslingerde. Zonde van al je werk toch!
* Dus, lever **minstens een half uur voor tijd in**. Download dan wat je hebt ingeleverd op Canvas. Geef het een andere naam om verwarring te voorkomen. En draai alle cellen, en bekijk het. Geen syntax fouten? Alle vragen gemaakt? Dan zit het vast wel goed, en hoef je niet in de zenuwen te zitten.

# Linked Data project

Maak een notebook met een end-to-end project gebaseerd op wikidata, en neem bijvoorbeeld deze als inspiratie:
* Mooie tutorial van begin tot eind met goede video: <https://media.ed.ac.uk/media/Wikidata+Sparql+Query+Tutorial/1_7v9v6s04> (Vrouwen educated in university of edinburg)
* Leuke tutorial met notebooks beschikbaar: <https://towardsdatascience.com/where-do-mayors-come-from-querying-wikidata-with-python-and-sparql-91f3c0af22e2>
* [Deze pagina](https://www.wikidata.org/wiki/Wikidata:WikiProject_every_politician/Netherlands/data/House/All_Members) over Nederlandse politici bevat een mooi begin, waar je nog heel wat moois van kan maken. 
* Of [deze](https://www.wikidata.org/wiki/Wikidata:WikiProject_Netherlands_Public_Libraries/Branches/By_Organisation) over bibliotheken. Ook hier kan je zowel het spreadsheet veel rijker maken, als natuurlijk op basis daarvan een geweldige interface maken.
    * Maar er is al een hoop mee gedaan hoor: zie bijvoorbeeld <https://www.wikidata.org/wiki/Wikidata:WikiProject_Netherlands_Public_Libraries/Maps#Country>
* <https://www.wikidata.org/wiki/Wikidata:WikiProject_COVID-19>


## Vereisten

1. Leuk probleem, goed uitgewerkt en gemotiveerd. Origineel, maar kan natuurlijk gebaseerd zijn op eerder werk.
    * Wees heel eerlijk met je inspiratiebronnen, en noem ze, en benoem exact waar jij verder bent gegaan (wat zijn jullie toevoegingen?)
2. Het moet gaan over **Nederlandse** data, of data over Nederland. 
    * Natuurlijk mag er ook een verband met het buitenland zijn.
2. Een dataset die je met SPARQL uit wikidata haalt, en die je netjes beschrijft, en aangeeft dat de data "klopt" met jouw probleem.
    * je dataset moet wel echt wat body hebben: dus flink wat rijen (items), en kolommen (eigenschappen)
    * je kan je data in spreadsheet formaat of SQL formaat of zelfs JSON formaat opslaan, en er dan later mee verder werken in Python-Pandas, SQL, of direct met python dicts.
3. Je laat nu op een aansprekende manier zien hoe jouw data jouw probleem oplost.
    * Dit kan prima in een jupyter notebook. Het is een **functioneel prototype** voor een mogelijke webapplicatie. 
    * Je kan [python widgets](https://ipywidgets.readthedocs.io/en/stable/examples/Widget%20Basics.html) gebruiken voor interactie met gebruiker
    * of plotly en dash voor gave interactieve plots en visuals (of seaborn als je het saai statisch wilt houden)
4. Schrijf een reflectie.
5. Jouw hele "blogpost/tutorial" is een Jupyter notebook die standalone perfect draait. (mits verbonden met het web).
6. Je plaatst je notebook op een publiek github adres, of ergens op Google colab. Met gedraaide output. Het is belangrijk dat anderen jouw notebook 
    * heel makkelijk kunen lezen
    * heel makkelijk kunnen downloaden, en runnen.
        * dus als je niet standaard modules gebruikt zet je in een aparte uitgecommentarieerde cel de `pip install commands`
7. En levert hem natuurlijk ook weer in via Canvas.

### Grading

* Ook hier weer via peer review binnen je groep.
* Plus een expert review van je tutor.
* Maak er wat moois, leerzaams, gaafs, en verrassends van! 

In [2]:
!pip install sparqlwrapper
# https://rdflib.github.io/sparqlwrapper/
!pip install folium
# https://python-visualization.github.io/folium/
!pip install plotly
# https://plotly.com/python/

import sys

# Het probleem
## Hoe presteren Nederlanders op de Olympische spelen?
We kregen een beetje de kriebels van het zelf niet meer mogen sporten door corona en dus waren we benieuwd hoe goed wij als land nou eigenlijk zijn in sport. Daarom kijken wij voor deze opdracht naar Nederlanders die ooit mee hebben gedaan aan een onderdeel van de Olympische spelen en goed hoe ze daar hebben gepresteerd. We kijken zowel naar de Zomerspelen als de Winterspelen. 

Om hier een antwoord op te krijgen, hebben we op wikidata alle participaties aan onderdelen van Olympische spelen opgezocht, waarin de participanten Nederlands zijn.

Per participatie halen we de volgende kenmerken op door middel van de SPARQL query hieronder:
- De participant
- Geboorteplaats en coördinaten van geboorteplaats van de participant
- Het soort Olympische Spelen (Zomer- of Winterspelen)
- De specifieke spelen waaraan is meegedaan (b.v '2000 Summer Olympics')
- Het specifieke sportonderdeel waaraan de participant heeft meegedaan (b.v 'cycling at the 1928 Summer Olympics – men's tandem')
- Het jaartal waarin de Spelen van de participatie plaatsvond
- Eventueel gewonnen medaille

In [2]:
from SPARQLWrapper import SPARQLWrapper, JSON

endpoint_url = "https://query.wikidata.org/sparql"

query = """SELECT DISTINCT  ?player ?playerLabel ?spelen ?spelenLabel ?spelenSpecifiek ?spelenSpecifiekLabel ?spelenJaar ?spelenOnderdeel ?spelenOnderdeelLabel  ?sport ?sportLabel ?medalLabel ?geboorteplaats ?geboorteplaatsLabel ?geboorteplaatsCoordinaten

WHERE{
  ?player wdt:P27 wd:Q29999;                                                          #?player -- country of citizenship -> Kingdom of the Netherlands
          wdt:P1344 ?spelenOnderdeel;                                                 #        -- participant in ---------> ?spelenOnderdeel
          wdt:P19 ?geboorteplaats.                                                    #        -- place of birth ---------> ?geboorteplaats
  ?geboorteplaats   wdt:P625 ?geboorteplaatsCoordinaten.                              #?geboorteplaats -- coordinate location -> ?geboorteplaatsCoordinaten
    
  ?spelen  wdt:P279 wd:Q5389. FILTER(?spelen = wd:Q159821 || ?spelen = wd:Q82414)     #?spelen -- subclass of --> Olympic Games  FILTER(?spelen = Summer Olympic Games || ?spelen = Winter Olympic Games)
  ?spelen wdt:P527 ?spelenSpecifiek.                                                  #?spelen -- has part -----> ?spelenSpecifiek
  
  ?spelenOnderdeel wdt:P31*/wdt:P361 wd:Q26132862;                                    #?spelenOnderdeel --instance of*/part of--> Olympic sports discipline event
                   wdt:P361* ?spelenSpecifiek;                                        #                 -- part of -------------> ?spelenSpecifiek
                   ?afgespeeldIn  ?spelenDatum;                                       #                 -- ?afgespeeldIn -------> ?spelenDatum
                   wdt:P641 ?sport                                                    #                 -- sport ---------------> ?sport
           
  FILTER(?afgespeeldIn = wdt:P585 || ?afgespeeldIn = wdt:P580).                       #FILTER(?afgespeeldIn = point in time || ?afgespeeldIn = start time)
                    
  BIND(YEAR(?spelenDatum) as ?spelenJaar).                                            
 
  OPTIONAL{ # medaille via property 'participant in' indien bestaand
    ?player p:P1344 ?participateStatement.                                            #?player -- participant in ---------------> ?participateStatement
    ?participateStatement ps:P1344 ?spel. FILTER(?spel = ?spelenOnderdeel).           #?participateStatement --participant in --> ?spel
    ?participateStatement pq:P166 ?medal.                                             #                      -- award received -> ?medal
  }
 
  SERVICE wikibase:label { bd:serviceParam wikibase:language "nl, en". }
}"""


## Resultaten ophalen en omzetten naar een dataframe

In [3]:
from pandas import json_normalize

def get_results(endpoint_url, query):
    sparql = SPARQLWrapper(endpoint_url)
    sparql.setQuery(query)
    sparql.setReturnFormat(JSON)

    result = sparql.query().convert()
    return json_normalize(result["results"]["bindings"])


resultsDF_original = get_results(endpoint_url, query)

## En hoe ziet het eruit? 

In [4]:
resultsDF_original.shape

(915, 39)

In [5]:
resultsDF_original.head()

geboorteplaats.value  ...  \
0   http://www.wikidata.org/entity/Q631499  ...   
1     http://www.wikidata.org/entity/Q9978  ...   
2  http://www.wikidata.org/entity/Q4681121  ...   
3      http://www.wikidata.org/entity/Q690  ...   
4    http://www.wikidata.org/entity/Q34370  ...   

                          spelenOnderdeelLabel.value sportLabel.xml:lang  \
0  canoeing at the 1988 Summer Olympics – women's...                  nl   
1  archery at the 2000 Summer Olympics – men's in...                  nl   
2  Schaatsen op de Olympische Winterspelen 2006 -...                  nl   
3  speed skating at the 1972 Winter Olympics – wo...                  nl   
4  swimming at the 1928 Summer Olympics – women's...                  nl   

  sportLabel.type          sportLabel.value medalLabel.xml:lang  \
0         literal        kanoën en kajakken                  nl   
1         literal              boogschieten                  nl   
2         literal        langebaanschaatsen                  nl   
3         literal  hardrijden op de schaats                  nl   
4         literal                 zwemsport                  nl   

  medalLabel.type              medalLabel.value geboorteplaatsLabel.xml:lang  \
0         literal   olympische bronzen medaille                           nl   
1         literal   olympische bronzen medaille                           nl   
2         literal   olympische bronzen medaille                           nl   
3         literal  olympische zilveren medaille                           nl   
4         literal  olympische zilveren medaille                           nl   

  geboorteplaatsLabel.type geboorteplaatsLabel.value  
0                  literal                   Beegden  
1                  literal                  Zaanstad  
2                  literal                    Goirle  
3                  literal                     Delft  
4                  literal                 Rotterdam  

[5 rows x 39 columns]

Oké, 915 resultaten en 39 kolommen (er zijn een hoop kolommen bijgekomen!). Ten eerste zijn die kolomnamen een beetje onhandig. Ook hebben we al die extra kolommen voor type en taal niet nodig. Veranderen dus!

In [6]:
resultsDF = resultsDF_original[[column for column in resultsDF_original.columns if ".value" in column]]
resultsDF = resultsDF.rename(columns = lambda col: col.replace(".value", ""))
resultsDF.head()

idata.org/entity/Q631499            Point(5.9208 51.1883)   
1     http://www.wikidata.org/entity/Q9978            Point(4.7767 52.4697)   
2  http://www.wikidata.org/entity/Q4681121            Point(5.0671 51.5203)   
3      http://www.wikidata.org/entity/Q690  Point(4.359444444 52.011944444)   
4    http://www.wikidata.org/entity/Q34370           Point(4.47917 51.9225)   

                                   spelen spelenJaar       playerLabel  \
0  http://www.wikidata.org/entity/Q159821       1988   Annemiek Derckx   
1  http://www.wikidata.org/entity/Q159821       2000  Wietse van Alten   
2   http://www.wikidata.org/entity/Q82414       2006        Ireen Wüst   
3   http://www.wikidata.org/entity/Q82414       1972      Stien Kaiser   
4  http://www.wikidata.org/entity/Q159821       1928       Marie Braun   

               spelenLabel          spelenSpecifiekLabel  \
0   Olympische Zomerspelen   Olympische Zomerspelen 1988   
1   Olympische Zomerspelen   Olympische Zomerspelen 2000   
2  Olympische Winterspelen  Olympische Winterspelen 2006   
3  Olympische Winterspelen  Olympische Winterspelen 1972   
4   Olympische Zomerspelen   Olympische Zomerspelen 1928   

                                spelenOnderdeelLabel  \
0  canoeing at the 1988 Summer Olympics – women's...   
1  archery at the 2000 Summer Olympics – men's in...   
2  Schaatsen op de Olympische Winterspelen 2006 -...   
3  speed skating at the 1972 Winter Olympics – wo...   
4  swimming at the 1928 Summer Olympics – women's...   

                 sportLabel                    medalLabel geboorteplaatsLabel  
0        kanoën en kajakken   olympische bronzen medaille             Beegden  
1              boogschieten   olympische bronzen medaille            Zaanstad  
2        langebaanschaatsen   olympische bronzen medaille              Goirle  
3  hardrijden op de schaats  olympische zilveren medaille               Delft  
4                 zwemsport  olympische zilveren medaille           Rotterdam

## Beter? Beter.
Maar oh wacht, voor de library die we zometeen gaan gebruiken, Folium, moeten de coördinaten omgedraaid en zonder text en haken zijn. Dus niet zoals hieronder.

In [7]:
resultsDF['geboorteplaatsCoordinaten'][0]

'Point(5.9208 51.1883)'

Ook aanpassen dus!

In [8]:
# Coordinaten draaien om ze bruikbaar te maken voor Folium zometeen
def prepareCoordinates(coordSeries):
    for i in range(len(coordSeries)):
        coordSeries[i] = coordSeries[i].strip('Point()')
        coordSeries[i] = [coord for coord in coordSeries[i].split(' ')]
        coordSeries[i][0] , coordSeries[i][1] = coordSeries[i][1] , coordSeries[i][0]
        coordSeries[i] = (str(coordSeries[i][0]), str(coordSeries[i][1]))

prepareCoordinates(resultsDF['geboorteplaatsCoordinaten'])

In [9]:
resultsDF['geboorteplaatsCoordinaten'][0]

('51.1883', '5.9208')

Kijk, dát kunnen we nou gebruiken.

Oké, en verder?

Nou, we willen op een kaart alle personen, de sport die ze beoefenen, hoeveel participaties ze hebben en hoeveel medailles ze hebben gewonnnen. Daarvoor kunnen we onze dataframe van eerder groeperen op personen (met bijhorende coördinaten), sport, participaties en medailles.

In [10]:
participatiesPerPersoon = resultsDF.groupby(['playerLabel', 'player', 'geboorteplaatsLabel', 'geboorteplaatsCoordinaten', 'sport', 'sportLabel']).count()[['spelen', 'medalLabel']]
participatiesPerPersoon.sort_values(by='medalLabel', ascending=False).head()

.org/entity/Q235022 Erp                 (51.6003, 5.6053)           http://www.wikidata.org/entity/Q902378   paardensport             8   
Inge de Bruijn            http://www.wikidata.org/entity/Q229491 Barendrecht         (51.85, 4.533333333)        http://www.wikidata.org/entity/Q31920    zwemsport                8   
Pieter van den Hoogenband http://www.wikidata.org/entity/Q298723 Maastricht          (50.866666666, 5.683333333) http://www.wikidata.org/entity/Q31920    zwemsport                7   
Sven Kramer               http://www.wikidata.org/entity/Q111320 Heerenveen          (52.95, 5.933333333)        http://www.wikidata.org/entity/Q17165362 langebaanschaatsen       6   

                                                                                                                                                                              medalLabel  
playerLabel               player                                 geboorteplaatsLabel geboorteplaatsCoordinaten   sport                                    sportLabel                      
Ireen Wüst                http://www.wikidata.org/entity/Q239546 Goirle              (51.5203, 5.0671)           http://www.wikidata.org/entity/Q17165362 langebaanschaatsen          11  
Anky van Grunsven         http://www.wikidata.org/entity/Q235022 Erp                 (51.6003, 5.6053)           http://www.wikidata.org/entity/Q902378   paardensport                 8  
Inge de Bruijn            http://www.wikidata.org/entity/Q229491 Barendrecht         (51.85, 4.533333333)        http://www.wikidata.org/entity/Q31920    zwemsport                    8  
Pieter van den Hoogenband http://www.wikidata.org/entity/Q298723 Maastricht          (50.866666666, 5.683333333) http://www.wikidata.org/entity/Q31920    zwemsport                    7  
Sven Kramer               http://www.wikidata.org/entity/Q111320 Heerenveen          (52.95, 5.933333333)        http://www.wikidata.org/entity/Q17165362 langebaanschaatsen           6

Vervolgens kunnen we met alleen de informatie uit deze dataframe en de library Folium een mooie interactieve kaart maken waarop alle personen uit de dataframe worden afgebeeld, met wat extra informatie.

In [11]:
import folium 
from folium.plugins import MarkerCluster

mapPersons = folium.Map(location=[52.370216, 4.895168], control_scale = True, zoom_start=10)
marker_clusterPersons = MarkerCluster().add_to(mapPersons)

for index, value in participatiesPerPersoon.iterrows():
    playerLabel, player, geboorteplaatsLabel, geboorteplaatsCoordinaten, sport, sportLabel, aantalParticipaties, aantalMedailles = index[0], index[1], index[2], index[3], index[4], index[5], value[0], value[1]

    popupPersons = folium.Popup(f"""   Speler: <br>
                                <a href={player}>{playerLabel}</a> 
                                <br><br>
                                Sport:<br> 
                                <a href={sport}>{sportLabel}</a>  
                                <br><br>
                                Participaties:<br> 
                                {aantalParticipaties} 
                                <br><br>
                                Aantal gewonnen Olympische medailles:<br>
                                {aantalMedailles}""", 
                                max_width = 200)
    folium.Marker(geboorteplaatsCoordinaten,popup= popupPersons,tooltip=playerLabel).add_to(marker_clusterPersons)


In [12]:
mapPersons

Gaaf! Maar er zijn al best wat Olympische Spelen geweest. In welke jaren zijn de meeste participanten geweest en de meeste medailles gewonnen?

Nou, daarvoor kunnen we onze dataframe groeperen op jaar!

In [13]:
perJaar = resultsDF.groupby(['spelenJaar']).count()[['player', 'medalLabel']].sort_values(by='spelenJaar', ascending=True)
perJaar.head()

player  medalLabel
spelenJaar                    
1900            22           6
1908             4           0
1920            29          16
1924            27          10
1928            35          14

En dan afbeelden in een mooie, interactieve grafiek via plotly.

Je kunt zelfs in de grafiek een vierkant slepen om daarop in te zoomen!

In [18]:
import plotly.graph_objs as go
participatieMedailles = go.Figure(
    data=[
        go.Bar(name='Aantal participaties', x=perJaar.index, y=perJaar['player'], width=2),
        go.Bar(name='Aantal gewonnen medailles', x=perJaar.index, y=perJaar['medalLabel'], width=1.5)
    ]
)
participatieMedailles.update_layout(
    autosize=False,
    width=1000,
    height=550,
    margin=dict(
        l=10,
        r=10,
        b=30,
        t=50,
    ),
    title_text='Aantal Nederlandse participaties en gewonnen medailles door de jaren heen',
    yaxis=dict(title_text='Aantal', range=[0, 80]),
    xaxis=dict(title_text='Jaartal', tickvals=perJaar.index, tickangle = -70)
)

Nou, hier is al best goed te zien dat wikidata niet de beste data heeft. Er is te zien dat de x-as inconsistenter wordt naarmate er verder terug in de tijd gekeken wordt. Maar oké.

Verder leek het ons leuk om te laten zien in welke  jaren een zelf gekozen aantal medailles gewonnen zijn, per sport. Hiervoor moeten we onze dataframe dus groeperen op jaar en sport.

In [19]:
sportPerJaar = resultsDF.groupby(['spelenJaar', 'sportLabel']).count()[['player', 'medalLabel']]
sportPerJaar.head()

player  medalLabel
spelenJaar sportLabel                        
1900       roeien              16           0
           schietsport          5           5
           zwemsport            1           1
1908       roeien               4           0
1920       baanwielrennen       5           5

Vervolgens gaan we widgets gebruiken! 

Gebruiksaanwijzing:

1. Run de cel hieronder om de slider te laden en te tonen.
2. Kies een getal op de slider
3. Run de cel onder de slider
4. Bekijk het resultaat!

In [20]:
import ipywidgets as widgets
from ipywidgets import HBox, Label, Layout

aantalMedailles = widgets.IntSlider(
    value=5,
    min=sportPerJaar['medalLabel'].min(),
    max=sportPerJaar['medalLabel'].max(),
    description='Minimum aantal gewonnen medailles:',
    layout=Layout(
        width='50%',
    ),
    style={'description_width': 'initial'}
)
display(aantalMedailles)

IntSlider(value=5, description='Minimum aantal gewonnen medailles:', layout=Layout(width='50%'), max=25, style…

In [21]:
from collections import defaultdict
q = [(i[0], i[1]) for i in sportPerJaar[sportPerJaar['medalLabel'] >= aantalMedailles.value].index]

sportMedailles = defaultdict(list)
for result in q:
    sportMedailles[result[1]].append(result[0])

for i in sportMedailles.items():
    if aantalMedailles.value == 1 and len(i[1]) > 1:
        print(f"Bij {i[0]} is in de jaren {i[1]} minimaal {aantalMedailles.value}  medaille gewonnen")
    elif len(i[1]) > 1:
        print(f"Bij {i[0]} zijn in de jaren {i[1]} minimaal {aantalMedailles.value}  medaille(s) gewonnen")
    else:
        print(f"Bij {i[0]} zijn in het jaar {i[1][0]} minimaal {aantalMedailles.value}  medaille(s) gewonnen")

Bij schietsport zijn in het jaar 1900 minimaal 5  medaille(s) gewonnen
Bij baanwielrennen zijn in het jaar 1920 minimaal 5  medaille(s) gewonnen
Bij boogschieten zijn in het jaar 1920 minimaal 5  medaille(s) gewonnen
Bij paardensport zijn in de jaren ['1924', '1928', '1996', '2000', '2012'] minimaal 5  medaille(s) gewonnen
Bij zwemsport zijn in de jaren ['1932', '1936', '1948', '1952', '1964', '1984', '2000', '2004', '2008', '2012'] minimaal 5  medaille(s) gewonnen
Bij hardrijden op de schaats zijn in de jaren ['1952', '1968', '1998', '2002', '2006'] minimaal 5  medaille(s) gewonnen
Bij wegwielrennen zijn in het jaar 1968 minimaal 5  medaille(s) gewonnen
Bij langebaanschaatsen zijn in de jaren ['1972', '1988', '2006', '2010', '2014'] minimaal 5  medaille(s) gewonnen


# Reflectie

Dit is de eerste keer dat we voor een project geheel met wikidata als informatiebron werken. Hierdoor zijn we erachter gekomen dat, hoe prachtig het idee van wikidata ook is, het ook erg onpraktisch kan zijn bij het gebruik van wikidata-data (ha) in een daadwerkelijk project. Omdat iedereen op elk moment pagina’s kan aanpassen en omdat wikidata grotendeels afhankelijk is van individuele mensen die hopelijk op een goede manier informatie invoeren, is wikidata als database eigenlijk best onbetrouwbaar. Een duidelijk voorbeeld hiervan, waar wij tegenaan liepen, is dat de ‘instance of’ statement van alle (x jaar) Winter Olympics was veranderd, waardoor onze query ineens geen winterspelen meer oppakte, terwijl het dat eerder wel deed. Ook mist veel informatie op wikidata, of wordt er op inconsequente manieren omgegaan met het invoeren van informatie. Veel mensen zijn bijvoorbeeld ‘participant of’ een ‘(x jaar) Summer Olympics’, terwijl andere mensen ‘participant of’ een specifiek onderdeel van een Spelen. In onze query kijken we alleen naar mensen bij wie wordt aangegeven dat ze direct hebben meegedaan aan een specifiek onderdeel van een Spelen, zodat ook duidelijk is welke sport ze hebben beoefend en waarvoor ze eventueel een medaille hebben gewonnen. Dit betekent echter wel dat veel participanten niet zijn meegenomen. Al met al hebben we veel geleerd door te werken met wikidata en beseffen we ook dat er nog veel te leren valt.